### PR TITLE
Add ability to get Workflow information

### PR DESCRIPTION
**Pull request by techncl**

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/dp/client/ProcessMonitorClient.scala [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+          .toMap
+      val relevantQueryParamsAsString: Map[String, String] = getMonitorsRequestAsMap.collect {
+        case (name, value) if value.iterator.nonEmpty => (name, value.iterator.mkString(","))
+      }
+
+      val getMonitorsUrl = uri"$apiBaseUrl/api/processmonitor/monitors?$relevantQueryParamsAsString"
+
+      for {
+        _ <-
+          if (!relevantQueryParamsAsString("name").startsWith("opex"))
```
Comment (MancunianSam) [inline src/main/scala/uk/gov/nationalarchives/dp/client/ProcessMonitorClient.scala:78]: You can do 
```scala
me.raiseWhen(!relevantQueryParamsAsString("name").startsWith("opex"))(PreservicaClientException(
            "The monitor name must start with 'opex'"
          ))
```
It's slightly less verbose I think.

File: src/main/scala/uk/gov/nationalarchives/dp/client/ProcessMonitorClient.scala [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    */
+  case class GetMonitorsRequest(
+      status: List[MonitorsStatus] = Nil,
+      name: Option[String] = None,
+      category: List[MonitorCategory] = Nil
+  ) {
+    val subcategory: List[String] = Nil
+  }
+
+  case class MonitorsError(code: String, description: String, status: String, file: String)
```
Comment (MancunianSam) [inline src/main/scala/uk/gov/nationalarchives/dp/client/ProcessMonitorClient.scala:124]: This case class isn't used anywhere. Do we need it?

**Review by MancunianSam — COMMENTED** [review]
Looks good, thanks. Just a couple of minor bits.

File: site-docs/src/main/paradox/processmonitors/usage/index.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+The process monitor client doesn't take a `Stream` type parameter as the `getMonitors` method doesn't need to stream. 
+
+@@snip [Clients.scala](../../../scala/examples/Clients.scala) { #process_monitor_client }
+
+@@include[method-heading.md](../../.includes/client/method-heading.md)
+
+### getMonitors
+* Generates the monitors input query from the arguments.
+* Sends this query in order to the `monitors` endpoint of the API.
```
[the remark this answers is not in the document]
Comment (techncl) [reply site-docs/src/main/paradox/processmonitors/usage/index.md:15]: Ah, no; this was a mistake

File: src/main/scala/uk/gov/nationalarchives/dp/client/ProcessMonitorClient.scala (hunk 2 above)
Comment (techncl) [reply src/main/scala/uk/gov/nationalarchives/dp/client/ProcessMonitorClient.scala:124]: Ah, no; this was a remnant of some other code.

File: src/main/scala/uk/gov/nationalarchives/dp/client/ProcessMonitorClient.scala (hunk 1 above)
Comment (techncl) [reply src/main/scala/uk/gov/nationalarchives/dp/client/ProcessMonitorClient.scala:78]: Ah, yes, forgot about that; this looks nicer

File: src/main/scala/uk/gov/nationalarchives/dp/client/Client.scala [hunk 4]
```diff
@@ -69,6 +69,27 @@ private[client] class Client[F[_], S](clientConfig: ClientConfig[F, S])(implicit
     }
   }
 
+  private[client] def sendJsonApiRequest[R](
+      url: String,
+      token: String,
+      method: Method,
+      requestBody: Option[String] = None,
+      responseSchema: ResponseAs[Either[ResponseException[String, Exception], R], Any]
```
[the remark this answers is not in the document]
Comment (techncl) [reply src/main/scala/uk/gov/nationalarchives/dp/client/Client.scala:77]: 👍. Yeah, that looks a lot cleaner. Thanks

**Review by MancunianSam — APPROVED** [review]
:+1:

**Pull request merged**